doc/dev/CustomRepoGuide.md

```diff
@@ -0,0 +1,387 @@
+
+As stated before, you are just executing bash commands, so the following section should be somehow self explanatory:
+
+~~~yml
+script:
+    - docker exec dock cmake -DCMAKE_CXX_COMPILER=$CXX -DCMAKE_BUILD_TYPE=$BUILD_TYPE -DCMAKE_CXX_FLAGS="$COVERAGE $WERROR" ../source
+    - docker exec dock make -j2
+    - docker exec dock ctest
+~~~
+ 
+The command `docker exec dock` just sends a bash command to the docker image that we named "dock". The only noteworthy thing is the cmake call, where we pass our environmental variables which we set in the `matrix:` section to cmake. Additionally you should have a look at the end of the cmake call: `../source`. Here we specify the path to our source files, which we mounted to "/home/nuto/source". Since we are using ".." our current location inside the image is obviously a folder located inside "/home/nuto". This folder is "build". This is the predefined starting directory of the current nuto image. You have to be careful which command you call from which directory. Calling a command from the wrong directory might cause your tests to fail. You can always check what you are doing inside the docker container by pulling the image to your computer and copying the commands from your .travis.yml. This will be discussed in the following section.
+
+
+
+**Additional notes**
+
+The travis .travis.yml can contain additional section that are not needed here. You can find an overview here: 
+
+https://docs.travis-ci.com/user/customizing-the-build/
+
+If you make any changes to your `.travis.yml` and want to know if it is valid before pushing to your repository, you can copy and paste the code into the corresponding field on 
+
+http://lint.travis-ci.org/
+
+Or follow the instructions here:
+
+https://docs.travis-ci.com/user/travis-lint
+
+# Docker
+
+Docker is a virtual machine that we use to run an image on travis. This way we avoid installing all the necessary components for NuTo over and over again during each test run on travis. This saves a lot of time but adds a little bit more complexity to the whole test environment. However, if your test fails and the reason is not obvious due to the output, you can easily reproduce the travis results on your own computer in order to find the exact issue. First you need to install docker
+
+~~~bash
+sudo apt-get install docker
```

Comment:
This does not install docker. 
~~~
$ apt-cache show docker | grep -A4 Description-en
Description-en: System tray for KDE3/GNOME2 docklet applications
 Docker is a docking application (WindowMaker dock app) which acts as a system
 tray for any desktop environment, allowing you to have a system tray without
 running the KDE/GNOME panel Docker was designed to work with Openbox 2, but it
 should work fine in any window manager.
~~~